pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go

```diff
@@ -419,8 +421,8 @@ func buildNetworkTopologyMetadataWithCDP(deviceID string, store *metadata.Store,
 
 func getRemDeviceAddressByCDPRemIndex(store *metadata.Store, strIndex string) string {
 	remoteDeviceAddressType := store.GetColumnAsString("cdp_remote.device_address_type", strIndex)
-	if remoteDeviceAddressType == "1" {
-		return store.GetColumnAsString("cdp_remote.device_address", strIndex)
+	if remoteDeviceAddressType == ciscoNetworkProtocolIPv4 {
+		return net.IP(store.GetColumnAsByteArray("cdp_remote.device_address", strIndex)).String()
```

Comment:
I think it's not an issue, but just wanted to note that if the bytes passed to net.IP() are not valid, it might return strings like `?AABBCCDD` (? followed by hex string).

Reply:
what use cases do you think of where we have invalid bytes ? like the device not reporting the correct value ?